### PR TITLE
Include network requests in FAQ

### PR DESCRIPTION
**Pull request by mschwager**

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -60,6 +60,16 @@ No. Semgrep CI runs fully in your CI pipeline and your source-code never leaves
...
 
+#### What network requests are made?
+Semgrep CI makes network requests in accordance with the data storage mentioned above.
+
+[Semgrep CLI](https://github.com/returntocorp/semgrep) makes the following network requests:
+
+* When running with `--generate-config`, Semgrep requests `https://raw.githubusercontent.com/returntocorp/semgrep-rules/develop/template.yaml` to obtain the latest configuration template.
+* When running without `--disable-version-check`, Semgrep requests `https://semgrep.dev/api/check-version` to check that it's running the latest version. Semgrep CLI is under rapid development. Running the latest version reduces our maintenance burden and ensures users have the latest bug fixes and features.
```
Comment (pabloest) [inline docs/faq.md:69]: Is there any data sent along with that request?

Comment (ievans) [reply docs/faq.md:69]: Mention this is cached so it will occur at most 1/x day @mschwager?

File: docs/faq.md [hunk 2]
```diff
@@ -60,6 +60,16 @@ No. Semgrep CI runs fully in your CI pipeline and your source-code never leaves
...
+* When providing a URL to `--output`, Semgrep performs an HTTP `POST` of the results to the specified URL.
+* When providing a registry ID like `p/security-audit` to `--config`, Semgrep requests the configuration from the [Semgrep Registry](https://semgrep.dev/explore).
```
Comment (ievans) [inline docs/faq.md:71]: Thanks for taking this on @mschwager. Some of these things are bit obvious and/or could quickly become stale -- I wouldn't necessarily be comprehensive, just say that configs can be fetched remotely which of course involves network lookup.

**Comment by mschwager** [comment]
Okay, I simplified and shortened everything in an attempt to keep it to a basic one-liner:


2021-01-29-113539_958x272_scrot

**Review by ievans — APPROVED** [review]
Love it--great work Matt!

**Pull request merged**